config/allowlist.go

```diff
@@ -54,39 +55,49 @@ type Allowlist struct {
 	// This targets the _secret_, not the content of the regex match like the
 	// Regexes slice.
 	StopWords []string
+
+	// commitMap is a normalized version of Commits, used for efficiency purposes.
+	commitMap    map[string]struct{}
+	regexPat     *regexp.Regexp
+	pathPat      *regexp.Regexp
+	stopwordTrie *ahocorasick.Trie
 }
 
 // CommitAllowed returns true if the commit is allowed to be ignored.
 func (a *Allowlist) CommitAllowed(c string) (bool, string) {
 	if c == "" {
 		return false, ""
 	}
-
-	for _, commit := range a.Commits {
-		if commit == c {
-			return true, c
-		}
+	if _, ok := a.commitMap[strings.ToLower(c)]; ok {
```

Comment:
@rgmz are there places today where it double scans commits?

Reply:
Not that I'm aware of. Using a map make sense here because it's `o(1)`[^1] versus iterating the slice `o(n)`.

[^1]: or something like this.